### PR TITLE
Support List of enums ar ArgumentInput types.

### LINKED ISSUE #644: bug: If the input is a list of enums, the deserialized list passed to the fetcher still contains strings.
If the input is a list of enums, the deserialized list passed to the fetcher still contains strings, e.g.:

```
Query {
  lookup(types: [MyTypeEnum!]): [Result]
}

enum MyTypeEnum {
  TYPE1,
  TYPE2
}
```

This will fail:
```
query TestQuery {
  lookup(types: [TYPE1])
}
```

However if you nest the enum inside another input object the deserialization does work correctly:
```
query {
  lookup(types: MyLookupType): [Result]
}

input MyLookupType {
  types: [MyTypeEnum!]!
}

enum MyTypeEnum {
  TYPE1,
  TYPE2
}
```

this will work, nested list of enums will be deserialized properly when calling the fetcher method:
```
Query TestQuery {
  lookup(types: { types: [TYPE1]})
}
```

### PR DESCRIPTION
**Pull request by berngp**
Pull Request type
----

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe):

Changes in this PR
----

This commit allows a list of enums to be used directly as an ArgumentInput.

Fixes #644

Out of Scope
----

This PR requires the developers to define a _Collection Type_, in this case the type of the Enum.

### CODE REVIEW

File: graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/internal/InputArgumentTest.kt [hunk 1]
```diff
@@ -1467,6 +1457,51 @@ internal class InputArgumentTest {
         verify { applicationContextMock.getBeansWithAnnotation(DgsComponent::class.java) }
     }
```
Comment (srinivasankavitha) [inline graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/internal/InputArgumentTest.kt:1459]: Do we already have a test case covering the nested enum type use case?

File: graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DataFetcherInvoker.kt [hunk 2]
```diff
@@ -239,8 +240,15 @@ class DataFetcherInvoker(
             } else {
                 InputObjectMapper.mapToJavaObject(parameterValue as Map<String, *>, targetType)
             }
-        } else if (parameter.type.isEnum && parameterValue !== null) {
-            (parameter.type.enumConstants as Array<Enum<*>>).find { it.name == parameterValue }
+        } else if ((parameter.type.isEnum || collectionType?.isEnum ?: false) && parameterValue != null) {
```
Comment (srinivasankavitha) [inline graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DataFetcherInvoker.kt:243]: What does this mean? `collectionType?.isEnum ?: false` Should this just be `collectionType?.isEnum`

Comment (berngp) [reply graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DataFetcherInvoker.kt:243]: It should just be `collectionType?.isEnum == true`, the Kotlin compiler doesn't like just `collectionType?.isEnum` 🤷

Comment (srinivasankavitha) [reply graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DataFetcherInvoker.kt:243]: Got it, I see.

**Review by srinivasankavitha — APPROVED** [review]

File: graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/internal/InputArgumentTest.kt (hunk 1 above)
Comment (berngp) [reply graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/internal/InputArgumentTest.kt:1459]: We have plenty of tests that include _enums_ as part of other complex types but not directly as an input argument.

**Pull request merged**